### PR TITLE
Modification to how documentation is constructed

### LINKED ISSUE #130: Update documentation
I propose we move away from restructured text (rst) files being written by hand. We can use `sphinx-apidoc` to generate rst files for all subpackages/modules. A slight modification to the index.rst will enable docs to use the programmatically generated API docs. We should keep the getting started and other helpful pieces of course (sphinx-apidoc is for api only).

### PR DESCRIPTION
**Pull request by avmarchenko**
resolves #130 

There are probably other ways to do this, and other styles for the docs. Open to input here so we can change it now, but could also do it down the line. PRing first to a branch specific to the issue. Then can PR from this branch to master when satisfied.

### CODE REVIEW

**Comment by tjduigna** [comment]
Playing around with it a bit more, it is confusing that we duplicate some toctree references in `index.rst` and `started.rst` where `index.rst` contains all the notebooks.  A flatter and less repetitive structure may be easier to peruse.  See incoming commit

**Comment by avmarchenko** [comment]
Awesome! Does this mean this PR might close two issues?

File: .travis.yml [hunk 1]
```diff
@@ -50,7 +50,7 @@ script:
     - if [[ $TRAVIS_OS_NAME == "linux" ]] && [[ $PYTHONVER == 3.6 ]]; then
           coverage run --source=exatomic setup.py test;
           rm -rf docs/source/*.txt;
-          sphinx-apidoc -eM -s txt -o docs/source/ exatomic/;
+          sphinx-apidoc -eM -s txt -o docs/source/ exatomic/ */tests/* ;
```
Comment (avmarchenko) [inline .travis.yml]: This is really a good idea. Can we be even more greedy and do `*test*`?

Comment (tjduigna) [reply .travis.yml]: Makes sense

**Comment by tjduigna** [comment]
I wonder if we should also prefer filtering out references with no docstrings (i think the autodoc setting of `:no-undoc-members:` rather than `:undoc-members:`).  It will thin out the toctree but mask areas in the source that are lacking documentation.  I am also curious whether we can filter out the metaclass definitions but that may require an extension or hack.

**Comment by avmarchenko** [comment]
Hey I've updated the code taking into account your comment about `:no-undoc-members:` (well I haven't changed it to that yet because...): I wanted to get feedback on what options you all think would be best to be included?

**Pull request merged**